### DISCUSSION TITLE
Is it possible to change ACL configuration on the fly?

### DISCUSSION THREAD

**Discussion by wodomierz** [processone/ejabberd#3745] [Questions] [answered]
I want to change configuration of ACL (adding, editing, removing restrictions for some user) on the fly.
In the docs I can only see how to setup ACL in the config file. Are the changes in the ACL section in the config file reloaded dynamically? Is there any other way to setup ACL, eg. using some db, which can be changed on the fly?

**Comment by prefiks** [comment, marked as answer]
Rules are updated from config file when you call 'ejabberdctr reload_config'. There is internal in memory database for acl rules, so theoretically you could update that, but i don't think there is external api to access it, that would require some custom code to perform changes.

**Reply by wodomierz** [reply]
Thanks!

**Comment by badlop** [comment]
> Are the changes in the ACL section in the config file reloaded dynamically?

As @prefiks explained, the [reload_config](https://docs.ejabberd.im/developer/ejabberd-api/admin-api/#reload-config) command will read your configuration file and apply it immediately. If you changed acl rules in the config files, they are applied as soon as you run that comand.

> Is there any other way to setup ACL, eg. using some db, which can be changed on the fly?

Yes, simply edit ejabberd.yml and run reload_config.

In case you plan to write/remove those acls programatically often, using a custom program, you may want to read [Include Additional Files](https://docs.ejabberd.im/admin/configuration/file-format/#include-additional-files). This way, your program only needs to edit a specific file called acl.yml and not mess with the big ejabberd.yml

**Reply by wodomierz** [reply]
Thanks!
